src/service.rs

```diff
@@ -1,45 +1,174 @@
+    }
+}
+
+impl Clear for GrpcMessage {
+    fn clear(&mut self) {
+        match self {
+            GrpcMessage::Auth(msg) => msg.clear(),
+            GrpcMessage::RateLimit(msg) => msg.clear(),
+        }
+    }
+}
+
+impl Message for GrpcMessage {
+    fn descriptor(&self) -> &'static MessageDescriptor {
+        match self {
+            GrpcMessage::Auth(msg) => msg.descriptor(),
+            GrpcMessage::RateLimit(msg) => msg.descriptor(),
+        }
+    }
+
+    fn is_initialized(&self) -> bool {
+        match self {
+            GrpcMessage::Auth(msg) => msg.is_initialized(),
+            GrpcMessage::RateLimit(msg) => msg.is_initialized(),
+        }
+    }
+
+    fn merge_from(&mut self, is: &mut CodedInputStream) -> ProtobufResult<()> {
+        match self {
+            GrpcMessage::Auth(msg) => msg.merge_from(is),
+            GrpcMessage::RateLimit(msg) => msg.merge_from(is),
+        }
+    }
+
+    fn write_to_with_cached_sizes(&self, os: &mut CodedOutputStream) -> ProtobufResult<()> {
+        match self {
+            GrpcMessage::Auth(msg) => msg.write_to_with_cached_sizes(os),
+            GrpcMessage::RateLimit(msg) => msg.write_to_with_cached_sizes(os),
+        }
+    }
+
+    fn write_to_bytes(&self) -> ProtobufResult<Vec<u8>> {
+        match self {
+            GrpcMessage::Auth(msg) => msg.write_to_bytes(),
+            GrpcMessage::RateLimit(msg) => msg.write_to_bytes(),
+        }
+    }
+
+    fn compute_size(&self) -> u32 {
+        match self {
+            GrpcMessage::Auth(msg) => msg.compute_size(),
+            GrpcMessage::RateLimit(msg) => msg.compute_size(),
+        }
+    }
+
+    fn get_cached_size(&self) -> u32 {
+        match self {
+            GrpcMessage::Auth(msg) => msg.get_cached_size(),
+            GrpcMessage::RateLimit(msg) => msg.get_cached_size(),
+        }
+    }
+
+    fn get_unknown_fields(&self) -> &UnknownFields {
+        match self {
+            GrpcMessage::Auth(msg) => msg.get_unknown_fields(),
+            GrpcMessage::RateLimit(msg) => msg.get_unknown_fields(),
+        }
+    }
+
+    fn mut_unknown_fields(&mut self) -> &mut UnknownFields {
+        match self {
+            GrpcMessage::Auth(msg) => msg.mut_unknown_fields(),
+            GrpcMessage::RateLimit(msg) => msg.mut_unknown_fields(),
+        }
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        match self {
+            GrpcMessage::Auth(msg) => msg.as_any(),
+            GrpcMessage::RateLimit(msg) => msg.as_any(),
+        }
+    }
+
+    fn new() -> Self
+    where
+        Self: Sized,
+    {
+        // Returning default value
+        GrpcMessage::default()
+    }
+
+    fn default_instance() -> &'static Self
+    where
+        Self: Sized,
+    {
+        #[allow(non_upper_case_globals)]
+        static instance: ::protobuf::rt::LazyV2<GrpcMessage> = ::protobuf::rt::LazyV2::INIT;
+        instance.get(|| GrpcMessage::RateLimit(RateLimitRequest::new()))
```

Comment:
The requirement for this confuses me... couldn't this "just" be a factory function and be passed around as a fn pointer instead? Not a big deal tho

Reply:
This one was to please the compiler, got the implementation from the message lib xD